forge-ai/src/main/java/forge/ai/ability/ConniveAi.java

```diff
@@ -0,0 +1,68 @@
+package forge.ai.ability;
+
+import forge.ai.ComputerUtil;
+import forge.ai.ComputerUtilCard;
+import forge.ai.SpellAbilityAi;
+import forge.game.ability.AbilityUtils;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.card.CardLists;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.game.zone.ZoneType;
+
+public class ConniveAi extends SpellAbilityAi {
+    @Override
+    protected boolean doTriggerAINoCost(Player ai, SpellAbility sa, boolean mandatory) {
+        final Card source = sa.getHostCard();
+        boolean preferred = true;
+        CardCollection list;
+        final int amount = AbilityUtils.calculateAmount(source, sa.getParamOrDefault("ConniveNum", "1"), sa);
```

Comment:
amount not used ?

Reply:
yeah, I think the discard part is already handled.